### PR TITLE
refactor and move create_temp_module into numba.tests.support

### LINKED ISSUE #6778: Refactor and move `create_temp_module` into `numba.tests.support`
Testing code `create_temp_module` from e.g. https://github.com/numba/numba/pull/6753#discussion_r584945784, exists in 3 different places, very slightly differently, this ticket is a reminder to refactor.

### PR DESCRIPTION
**Pull request by Alexander-Makaryev**
Resolves #6778

This is a refactoring of code duplication.

Please look at issue for more details.

### CODE REVIEW

File: numba/tests/support.py [hunk 1]
```diff
@@ -878,3 +880,38 @@ def check_warnings(self, messages, category=RuntimeWarning):
...
+    out = []
+    for key, value in jit_options.items():
+        if isinstance(value, str):
+            value = '"{}"'.format(value)
+        out.append('{}={}'.format(key, value))
+    return ', '.join(out)
+
+
+@contextlib.contextmanager
+def create_temp_module(source_lines, **jit_options):
```
Comment (stuartarchibald) [inline numba/tests/support.py:897]: Suggest adding a doc string now this is a formal utility.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for doing this refactoring, patch looks good, just one suggestion about adding a docstring else good to go.

File: numba/tests/support.py (hunk 1 above)
Comment (Alexander-Makaryev) [reply numba/tests/support.py:897]: @stuartarchibald 
done
thanks

**Review by stuartarchibald — APPROVED** [review]
Many thanks for the patch and fixes!

File: numba/tests/support.py (hunk 1 above)
Comment (stuartarchibald) [reply numba/tests/support.py:897]: thanks

**Pull request merged**